pkg/field/validation.go

```diff
@@ -0,0 +1,108 @@
+package field
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/spf13/viper"
+)
+
+type ErrConfigurationMissingFields struct {
+	errors []error
+}
+
+func (e *ErrConfigurationMissingFields) Error() string {
+	var messages []string
+
+	for _, err := range e.errors {
+		messages = append(messages, err.Error())
+	}
+
+	return fmt.Sprintf("errors found:\n%s", strings.Join(messages, "\n"))
+}
+
+func (e *ErrConfigurationMissingFields) Push(err error) {
+	e.errors = append(e.errors, err)
+}
+
+// Validate perform validation of field requirement and constraints
+// relationships after the configuration is read.
+// We don't check the following:
+//   - if required fields are mutually exclusive
+//   - repeated fields (by name) are defined
+//   - if sets of fields are mutually exclusive and required
+//     together at the same time
+func Validate(c Configuration, v *viper.Viper) error {
+	present := make(map[string]int)
+	missingFieldsError := &ErrConfigurationMissingFields{}
+
+	// check if required fields are present
+	for _, f := range c.Fields {
+		isNonZero := false
+		switch f.FieldType {
+		case reflect.Bool:
+			isNonZero = v.GetBool(f.FieldName)
```

Comment:
I made `--insecure-skip-verify` required in baton-ldap to test this out. I have to set it to true or else it errors with `field insecure-skip-verify of type bool is marked as required but it has a zero-value`

Also we should probably print out whether fields are required or not in the help text.

Reply:
> I made `--insecure-skip-verify` required in baton-ldap to test this out. I have to set it to true or else it errors with `field insecure-skip-verify of type bool is marked as required but it has a zero-value`

The problem with this is that Viper does not return pointer values, like `*bool`, giving us a third option in such case. We only have the zero-value of each type to know if the flag/envvar was set or not.

Reply:
> Also we should probably print out whether fields are required or not in the help text.

Implemented

Reply:
Considering these restrictions, I think we just just not allow bools to be required. Otherwise "required" just means "it must be true", in which case... why even have the bool argument?

Reply:
indeed

I'll add that restriction